agent/agent.go

```diff
@@ -697,13 +712,15 @@ func stopServiceInputs(inputs []*models.RunningInput) {
 }
 
 // StopInputPlugin stops an input plugin
-func (a *Agent) StopInputPlugin(input string) error {
+func (a *Agent) StopInputPlugin(input string, shouldUpdatePlugin bool) error {
```

Comment:
In what case would this function be called and ```shouldUpdatePlugin``` is false?

Reply:
`StopInputPlugin` is called in `UpdateInputPlugin` before it is updated (but we don't want to modify the config in that case)

Reply:
Why wouldn't we modify the config if the plugin is updated?